### ISSUE TITLE
Warning when running `mix deps.get` and no requirement provided

### ISSUE THREAD

**Issue by eksperimental** [elixir-lang/elixir#8555]
```elixir
# mix.exs
  defp deps do
    [
      {:ex_doc, only: :dev, runtime: false},
    ]
  end
```

```console
~/git/eksperimental/testing$ mix  deps.get
ex_doc is missing its version requirement, use ">= 0.0.0" if it should match any version
Running dependency resolution...
Dependency resolution completed:
  earmark 1.3.0
  ex_doc 0.19.1
  ...
```

I shouldn't get that warning, since `mix help deps` states 

> 
>   Dependencies must be specified in the mix.exs file in one of the following
>   formats:
> 
>       {app, requirement}
>       {app, opts}
>       {app, requirement, opts}
> 
>   Where:
> 
>     • app is an atom
>     • requirement is a Version requirement or a regular expression
>     • opts is a keyword list of options

We either need to remove the warning, or update the help information for the `deps` task.

**Comment by wojtekmach** [comment]
This is `Mix.SCM` specific so I believe the docs are correct as is.

For example it's allowed to do:

```
{:ex_doc, github: "elixir-lang/ex_doc"}
```

Hex, via `Hex.SCM`, requires the version requirement to be present and the error comes from there.

**Comment by eksperimental** [comment]
true, I cannnot find that message in Elixir core, but we have a similar situation:

https://github.com/elixir-lang/elixir/blob/2e576f67a00164d88e2e242e523dc3b5c1a8a5db/lib/mix/lib/mix/dep/loader.ex#L254-L272

**Comment by eksperimental** [comment]
> true, I cannnot find that message in Elixir core, but we have a similar situation:
> 
> [elixir/lib/mix/lib/mix/dep/loader.ex](https://github.com/elixir-lang/elixir/blob/2e576f67a00164d88e2e242e523dc3b5c1a8a5db/lib/mix/lib/mix/dep/loader.ex#L254-L272)
> 
> Lines 254 to 272 in [2e576f6](/elixir-lang/elixir/commit/2e576f67a00164d88e2e242e523dc3b5c1a8a5db)
> 
>    defp invalid_dep_format(dep) do 
>      Mix.raise(""" 
>      Dependency specified in the wrong format: 
>   
>          #{inspect(dep)} 
>   
>      Expected: 
>   
>          {app, opts} | {app, requirement} | {app, requirement, opts} 
>   
>      Where: 
>   
>          app :: atom 
>          requirement :: String.t | Regex.t 
>          opts :: keyword 
>   
>      If you want to skip the requirement (not recommended), use ">= 0.0.0". 
>      """) 
>    end

@josevalim when does this happen? The situation is similar to the one in Hex and should be documented IMO

**Comment by josevalim** [comment]
It is OK for SCMs to impose extra requirements that Mix can’t because Mix
is meant to work with multiple SCMs. So Hex needs to improve the error
messages for  regexes here.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*
